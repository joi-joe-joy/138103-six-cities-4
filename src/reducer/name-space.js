export const NameSpace = {
  DATA: `DATA`,
  USER: `USER`,
  PLACE: `PLACE`
};
