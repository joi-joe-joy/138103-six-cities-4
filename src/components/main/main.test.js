import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {NameSpace} from "../../reducer/name-space.js";
import Main from "./main";

const mockStore = configureStore([]);

const offers = [
  {
    bedrooms: 2,
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Air conditioning`, `Laptop friendly workspace`, `Baby seat`, `Fridge`, `Breakfast`, `Washer`, `Washing machine`, `Dishwasher`, `Coffee machine`, `Towels`],
    host: {id: 25, name: `Angelina`, isPro: true, avatarUrl: `img/avatar-angelina.jpg`},
    id: 1,
    images: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`],
    isFavorite: false,
    isPremium: true,
    location: {
      latitude: 48.865610000000004,
      longitude: 2.350499,
      zoom: 16
    },
    maxAdults: 8,
    previewImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
    price: 397,
    rating: 3.6,
    title: `Penthouse, 4-5 rooms + 5 balconies`,
    type: `hotel`
  }
];

describe(`Render Main`, () => {
  it(`Main with offers`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        city: {
          name: `Paris`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13
          }
        },
        offers,
        citiesList: [{
          name: `Paris`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13
          }
        }, {
          name: `Amsterdam`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13
          }
        }]
      },
      [NameSpace.PLACE]: {
        sorting: `popular`,
        hoverCard: offers[0]
      },
      [NameSpace.USER]: {
        authorizationStatus: `AUTH`
      },
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <Main
              offers={offers}
              onLocationClick={()=>{}}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Main without offers`, () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        city: {
          name: `Paris`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13
          }
        },
        offers,
        citiesList: [{
          name: `Paris`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13
          }
        }, {
          name: `Amsterdam`,
          location: {
            latitude: 48.85661,
            longitude: 2.351499,
            zoom: 13
          }
        }]
      },
      [NameSpace.PLACE]: {
        sorting: `popular`,
        hoverCard: offers[0]
      },
      [NameSpace.USER]: {
        authorizationStatus: `AUTH`
      },
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <Main
              onLocationClick={()=>{}}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
