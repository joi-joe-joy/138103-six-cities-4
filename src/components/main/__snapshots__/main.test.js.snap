// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main Main with offers 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Paris
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             places to stay in 
            Paris
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                height="4"
                style={
                  Object {
                    "marginBottom": "1px",
                    "marginLeft": "5px",
                  }
                }
                width="7"
              />
            </span>
            <ul
              className="places__options places__options--custom"
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="places__list cities__places-list tabs__content"
          >
            <article
              className="place-card cities__place-card"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Penthouse, 4-5 rooms + 5 balconies"
                    className="place-card__image"
                    height="200"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      397
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      height="18"
                      width="17"
                    />
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    onClick={[Function]}
                  >
                    Penthouse, 4-5 rooms + 5 balconies
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Hotel
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="map cities__map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Render Main Main without offers 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Paris
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             places to stay in 
            Paris
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              className="places__sorting-type"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
              <svg
                height="4"
                style={
                  Object {
                    "marginBottom": "1px",
                    "marginLeft": "5px",
                  }
                }
                width="7"
              />
            </span>
            <ul
              className="places__options places__options--custom"
            >
              <li
                className="places__option places__option--active"
                onClick={[Function]}
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option"
                onClick={[Function]}
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="places__list cities__places-list tabs__content"
          >
            <article
              className="place-card cities__place-card"
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              <div
                className="place-card__image-wrapper cities__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Penthouse, 4-5 rooms + 5 balconies"
                    className="place-card__image"
                    height="200"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      397
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      height="18"
                      width="17"
                    />
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    onClick={[Function]}
                  >
                    Penthouse, 4-5 rooms + 5 balconies
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Hotel
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="map cities__map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
